File: killrvideo-service-videocatalog/src/main/java/com/killrvideo/service/video/request/GetUserVideoPreviewsRequestData.java
package com.killrvideo.service.video.request;

import lombok.AllArgsConstructor;
import lombok.Getter;

import java.time.Instant;
import java.util.Optional;
import java.util.UUID;

@SuppressWarnings("OptionalUsedAsFieldOrParameterType")
@Getter
@AllArgsConstructor
public class GetUserVideoPreviewsRequestData {
    private final UUID userId;
    private final Optional<UUID> startingVideoId;
    private final Optional<Instant> startingAddedDate;
    private final Optional<Integer> pagingSize;
    private final Optional<String> pagingState;

    public GetUserVideoPreviewsRequestData(UUID userId) {
        this(
                userId,
                Optional.empty(),
                Optional.empty(),
                Optional.empty(),
                Optional.empty());
    }
}

File: killrvideo-service-videocatalog/src/main/kotlin/com/killrvideo/service/video/request/GetUserVideoPreviewsRequestData.kt
package com.killrvideo.service.video.request

import java.time.Instant
import java.util.*

data class GetUserVideoPreviewsRequestData(
    val userId: UUID,
    val startingVideoId: UUID?,
    val startingAddedDate: Instant?,
    val pagingSize: Int?,
    val pagingState: String?
) {
    constructor(userId: UUID) : this(
        userId=userId,
        startingVideoId = null,
        startingAddedDate = null,
        pagingSize = null,
        pagingState = null
    )
}
